### PR TITLE
Account policy

### PR DESCRIPTION
**Pull request by jcwalker**
Added AccountPolicy resource

---
This change is [Reviewable](https://reviewable.io/reviews/powershell/securitypolicydsc/58)

### CODE REVIEW

File: DSCResources/MSFT_AccountPolicy/MSFT_AccountPolicy.psm1 [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+            else
+            {
+                $kerberosPolicies += "$($policyData.Value)=$newValue"
+            }
+        }
+    }
+
+    $infTemplate = Add-PolicyOption -SystemAccessPolicies $systemAccessPolicies -KerberosPolicies $registryPolicies
+    
+    Out-File -InputObject $infTemplate -FilePath $accountPolicyToAddInf -Encoding unicode -Force
```
Comment (mcollera) [inline DSCResources/MSFT_AccountPolicy/MSFT_AccountPolicy.psm1]: Should we remove file when we are done?

File: DSCResources/SecurityPolicyResourceHelper/SecurityPolicyResourceHelper.psm1 [hunk 2]
```diff
@@ -370,3 +370,109 @@ function ConvertTo-Sid
...
+
+
+<#
+    .SYNOPSIS
+        Retrieves the Security Option Data to map the policy name and values as they appear in the Security Template Snap-in
+
+    .PARAMETER FilePath
+        Path to the file containing the Security Option Data
+#>
+function Get-PolicyOptionData
```
Comment (mcollera) [inline DSCResources/SecurityPolicyResourceHelper/SecurityPolicyResourceHelper.psm1]: Is this function supposed to do something?

**Comment by athaynes** [comment]
:lgtm:

---

Review status: 0 of 22 files reviewed at latest revision, 8 unresolved discussions.

---

*[DSCResources/MSFT_AccountPolicy/AccountPolicyData.psd1, line 8 at r2](https://reviewable.io:443/reviews/powershell/securitypolicydsc/58#-KutHdZBrX_8FoT2_K04:-KutHdZBrX_8FoT2_K05:b5c0dtv) ([raw file](https://github.com/powershell/securitypolicydsc/blob/13b878915b3f6d9425a26ffe2dd2e249c5b10eb7/DSCResources/MSFT_AccountPolicy/AccountPolicyData.psd1#L8)):*
> ```PowerShell
>         Option  = @{
>             String = ''
>         }        
> ```

trailing spaces after every option table in this file

---

*[DSCResources/MSFT_AccountPolicy/MSFT_AccountPolicy.psm1, line 51 at r2](https://reviewable.io:443/reviews/powershell/securitypolicydsc/58#-KutHaV15XveqxPlBKS6:-KutHaV2x7FVbc9-592K:b88ta8c) ([raw file](https://github.com/powershell/securitypolicydsc/blob/13b878915b3f6d9425a26ffe2dd2e249c5b10eb7/DSCResources/MSFT_AccountPolicy/MSFT_AccountPolicy.psm1#L51)):*
> ```PowerShell
>             {
>                 $resultValue = ( $accountPolicyData.$accountPolicy.Option.GetEnumerator() | 
>                     Where-Object -Property Value -eq $currentValue.Trim() ).Name             
> ```

trailing spaces

---

*[DSCResources/MSFT_AccountPolicy/MSFT_AccountPolicy.psm1, line 57 at r2](https://reviewable.io:443/reviews/powershell/securitypolicydsc/58#-KutHZnbHrE2Hwp2nLDV:-KutHZnbHrE2Hwp2nLDW:b88ta8c) ([raw file](https://github.com/powershell/securitypolicydsc/blob/13b878915b3f6d9425a26ffe2dd2e249c5b10eb7/DSCResources/MSFT_AccountPolicy/MSFT_AccountPolicy.psm1#L57)):*
> ```PowerShell
>                 $resultValue = $null
>             }
>         }        
> ```

trailing spaces

---

*[DSCResources/MSFT_AccountPolicy/MSFT_AccountPolicy.psm1, line 298 at r2](https://reviewable.io:443/reviews/powershell/securitypolicydsc/58#-KutHLVB9LIRfEmlWxf-:-KutHLVB9LIRfEmlWxf0:b88ta8c) ([raw file](https://github.com/powershell/securitypolicydsc/blob/13b878915b3f6d9425a26ffe2dd2e249c5b10eb7/DSCResources/MSFT_AccountPolicy/MSFT_AccountPolicy.psm1#L298)):*
> ```PowerShell
>             Write-Verbose -Message ( $script:localizedData.TestingPolicy -f $policy )
>             Write-Verbose -Message ( $script:localizedData.PoliciesBeingCompared -f $($currentAccountPolicies[$policy] -join ',' ), $($desiredAccountPolicies[$policy] -join ',' ) )
>                        
> ```

trailing spaces

---

*[DSCResources/MSFT_AccountPolicy/MSFT_AccountPolicy.psm1, line 302 at r2](https://reviewable.io:443/reviews/powershell/securitypolicydsc/58#-KutHPq8tP_jJVgrpDYw:-KutHPq9-iyHXCCrYbj_:b88ta8c) ([raw file](https://github.com/powershell/securitypolicydsc/blob/13b878915b3f6d9425a26ffe2dd2e249c5b10eb7/DSCResources/MSFT_AccountPolicy/MSFT_AccountPolicy.psm1#L302)):*
> ```PowerShell
>             {
>                 return $false
>             }                    
> ```

trailing spaces

---

*[DSCResources/SecurityPolicyResourceHelper/SecurityPolicyResourceHelper.psm1, line 406 at r2](https://reviewable.io:443/reviews/powershell/securitypolicydsc/58#-KutH7ib8U6LQSp9mMH4:-KutH7icYIs8iGoZem5-:b-4gc3yq) ([raw file](https://github.com/powershell/securitypolicydsc/blob/13b878915b3f6d9425a26ffe2dd2e249c5b10eb7/DSCResources/SecurityPolicyResourceHelper/SecurityPolicyResourceHelper.psm1#L406)):*
> ```PowerShell
>     Param
>     (
>         [Parameter( Mandatory = $true)]
> ```

random space

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/securitypolicydsc/58#-:-KutHvPw_743C4L4n-C6:bnfp4nl)*

**Comment by jcwalker** [comment]
Review status: 0 of 22 files reviewed at latest revision, 8 unresolved discussions.

---

*[DSCResources/MSFT_AccountPolicy/AccountPolicyData.psd1, line 8 at r2](https://reviewable.io:443/reviews/powershell/securitypolicydsc/58#-KutHdZBrX_8FoT2_K04:-Kv9odwiQpkCRBTwSXXV:b-896fix) ([raw file](https://github.com/powershell/securitypolicydsc/blob/13b878915b3f6d9425a26ffe2dd2e249c5b10eb7/DSCResources/MSFT_AccountPolicy/AccountPolicyData.psd1#L8)):*
<details><summary><i>Previously, athaynes (Adam Haynes) wrote…</i></summary><blockquote>

trailing spaces after every option table in this file
</blockquote></details>

Done.

---

*[DSCResources/MSFT_AccountPolicy/MSFT_AccountPolicy.psm1, line 51 at r2](https://reviewable.io:443/reviews/powershell/securitypolicydsc/58#-KutHaV15XveqxPlBKS6:-Kv9pB8oEWwZKuve-Sok:b-896fix) ([raw file](https://github.com/powershell/securitypolicydsc/blob/13b878915b3f6d9425a26ffe2dd2e249c5b10eb7/DSCResources/MSFT_AccountPolicy/MSFT_AccountPolicy.psm1#L51)):*
<details><summary><i>Previously, athaynes (Adam Haynes) wrote…</i></summary><blockquote>

trailing spaces
</blockquote></details>

Done.

---

*[DSCResources/MSFT_AccountPolicy/MSFT_AccountPolicy.psm1, line 57 at r2](https://reviewable.io:443/reviews/powershell/securitypolicydsc/58#-KutHZnbHrE2Hwp2nLDV:-Kv9pwFrxF64-jZOeC7S:b-896fix) ([raw file](https://github.com/powershell/securitypolicydsc/blob/13b878915b3f6d9425a26ffe2dd2e249c5b10eb7/DSCResources/MSFT_AccountPolicy/MSFT_AccountPolicy.psm1#L57)):*
<details><summary><i>Previously, athaynes (Adam Haynes) wrote…</i></summary><blockquote>

trailing spaces
</blockquote></details>

Done.

---

*[DSCResources/MSFT_AccountPolicy/MSFT_AccountPolicy.psm1, line 195 at r2](https://reviewable.io:443/reviews/powershell/securitypolicydsc/58#-KukY9yWcX0rkTikgsOi-r2-195:-Kv9qpN38tIidnU6Gzyv:b-896fix) ([raw file](https://github.com/powershell/securitypolicydsc/blob/13b878915b3f6d9425a26ffe2dd2e249c5b10eb7/DSCResources/MSFT_AccountPolicy/MSFT_AccountPolicy.psm1#L195)):*
<details><summary><i>Previously, mcollera wrote…</i></summary><blockquote>

Should we remove file when we are done?
</blockquote></details>

Done.

---

*[DSCResources/MSFT_AccountPolicy/MSFT_AccountPolicy.psm1, line 298 at r2](https://reviewable.io:443/reviews/powershell/securitypolicydsc/58#-KutHLVB9LIRfEmlWxf-:-Kv9qscYWSqdl2xBe3Sj:b-896fix) ([raw file](https://github.com/powershell/securitypolicydsc/blob/13b878915b3f6d9425a26ffe2dd2e249c5b10eb7/DSCResources/MSFT_AccountPolicy/MSFT_AccountPolicy.psm1#L298)):*
<details><summary><i>Previously, athaynes (Adam Haynes) wrote…</i></summary><blockquote>

trailing spaces
</blockquote></details>

Done.

---

*[DSCResources/MSFT_AccountPolicy/MSFT_AccountPolicy.psm1, line 302 at r2](https://reviewable.io:443/reviews/powershell/securitypolicydsc/58#-KutHPq8tP_jJVgrpDYw:-Kv9r-TA_hNvXVG4Zf4N:b-896fix) ([raw file](https://github.com/powershell/securitypolicydsc/blob/13b878915b3f6d9425a26ffe2dd2e249c5b10eb7/DSCResources/MSFT_AccountPolicy/MSFT_AccountPolicy.psm1#L302)):*
<details><summary><i>Previously, athaynes (Adam Haynes) wrote…</i></summary><blockquote>

trailing spaces
</blockquote></details>

Done.

---

*[DSCResources/SecurityPolicyResourceHelper/SecurityPolicyResourceHelper.psm1, line 382 at r2](https://reviewable.io:443/reviews/powershell/securitypolicydsc/58#-KukY9yWcX0rkTikgsOm-r2-382:-Kv9tqf29UnqW4K4IEe8:b-896fix) ([raw file](https://github.com/powershell/securitypolicydsc/blob/13b878915b3f6d9425a26ffe2dd2e249c5b10eb7/DSCResources/SecurityPolicyResourceHelper/SecurityPolicyResourceHelper.psm1#L382)):*
<details><summary><i>Previously, mcollera wrote…</i></summary><blockquote>

Is this function supposed to do something?
</blockquote></details>

Done.

---

*[DSCResources/SecurityPolicyResourceHelper/SecurityPolicyResourceHelper.psm1, line 406 at r2](https://reviewable.io:443/reviews/powershell/securitypolicydsc/58#-KutH7ib8U6LQSp9mMH4:-Kv9twIJursS0Y_g7zD6:b-896fix) ([raw file](https://github.com/powershell/securitypolicydsc/blob/13b878915b3f6d9425a26ffe2dd2e249c5b10eb7/DSCResources/SecurityPolicyResourceHelper/SecurityPolicyResourceHelper.psm1#L406)):*
<details><summary><i>Previously, athaynes (Adam Haynes) wrote…</i></summary><blockquote>

random space
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/securitypolicydsc/58)*

**Pull request merged**